packages/server/src/index.ts

```diff
@@ -89,6 +121,96 @@ export default class Server {
 
     this.serverAndClient.addMethod(method, handler);
   }
+
+  /**
+   * Set the instance [[currencies]] object
+   *
+   * @param params.walletCurrencies - the list of all the currencies handled by
+   * the wallet
+   * @param params.allowedCurrenciesFilters - this list of allowed currencies
+   * filters for a specific Wallet APP, extracted from it's `categories`
+   * manifest field.
+   * These filters must be glob
+   * https://en.wikipedia.org/wiki/Glob_(programming)
+   */
+  setCurrencies({
+    walletCurrencies,
+    allowedCurrenciesFilters,
+  }: {
+    walletCurrencies: Currency[];
+    allowedCurrenciesFilters: "*" | string[];
+  }): void {
+    // If the Wallet APP has access to all available currencies (i.e: no restriction)
+    if (allowedCurrenciesFilters === "*") {
+      this.currencies = walletCurrencies;
+      return;
+    }
```

Comment:
I think we agreed that we don't need to support `*` in the filters

Reply:
I think we agreed the we don't need to support `*` filter for the API RPC methods (like `account.list` or `currency.list`).

For the server setter methods, we could decide not to support this filter but in this case either the Wallet will need to support this `*` filter **or** we would need to update our manifest spec to make the `currencies` property optional (which is not the case today, hence the `array of strings (string[]) or "*"` type, cf. [here](https://github.com/LedgerHQ/wallet-api/blob/main/spec/core/manifest.md)).

I would rather keep this logic as is and compatible with our current manifest spec / API for this first iteration and update this in a second time (depending on the chosen solution)